src/cargo/ops/cargo_new.rs

```diff
@@ -258,6 +258,13 @@ fn check_name(
             name
         ))?;
     }
+    if name != name.to_lowercase() {
+        shell.warn(format!(
+            "the name `{}` does not follow the preferred snake case or kebab case format\n\
+            See https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#General-naming-conventions",
```

Comment:
1. Should we just say "lower case" since that is simpler and the only thing decided at this time?
2. Mirroring the message from rustc might be a good start "package `{name}` should have a lowercase name such as `{lower_name}`"
3. We likely shouldn't be linking to that RFC.  It refers to "crate" and not 'package".  Sometimes those get mixed up but the [current naming guidelines, which also conflate crate and package, say there is no consesus](https://rust-lang.github.io/api-guidelines/naming.html)

Reply:
Wanted to keep the warning message similar to other warnings messages in the file.